fluss-client/src/main/java/com/alibaba/fluss/client/write/Sender.java

```diff
@@ -172,13 +173,25 @@ public void run() {
     public void runOnce() throws Exception {
         if (idempotenceManager.idempotenceEnabled()) {
             // may be wait for writer id.
-            idempotenceManager.maybeWaitForWriterId();
+            idempotenceManager.maybeWaitForWriterId(prepareInitWriterRequest());
```

Comment:
We can move the construction of `InitWriterRequest` into `IdempotenceManager` when we need to send the request (`maybeWaitForWriterId` only accepts a `Set<PhysicalTablePath>` parameter). Currently, it's an overhead to construct the request for each run.